client/src/Context/AuthContext.js

```diff
@@ -1,44 +1,49 @@
-import React, { createContext, useContext, useEffect, useState } from 'react'
-import { getAllClientData } from '../utils/ApiUtils';
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { getAllClientData } from "../utils/ApiUtils";
 
 const clientContext = createContext();
 
 function AuthContext({ children }) {
-    const [isLogin, setIsLogin] = useState(localStorage.getItem("token") ? true : false);
-    const [clientData, setClientData] = useState();
-    const [token, setToken] = useState();
+  const [isLogin, setIsLogin] = useState(
+    localStorage.getItem("token") ? true : false
+  );
+  const [clientData, setClientData] = useState();
+  const [token, setToken] = useState();
 
-    const fetchClientData = () => {
-        if (isLogin) {
-            // localStorage.setItem("token", JSON.stringify(res.data.token));
-            const temp = JSON.parse(localStorage.getItem("token"))
-            // console.log(temp)
-            setToken(temp);
-            getAllClientData(temp)
-                .then((res) => {
-                    console.log(res.data.msg);
-                    setClientData(res.data.msg);
-                })
-                .catch((error) => {
-                    console.log(error);
-                })
-        }
-        else {
-            setToken();
-            setClientData();
-            localStorage.removeItem("token");
-        }
-        // console.log("clientData fetched");
+  const fetchClientData = () => {
+    if (isLogin) {
+      // localStorage.setItem("token", JSON.stringify(res.data.token));
+      const temp = JSON.parse(localStorage.getItem("token"));
+      // console.log(temp)
+      setToken(temp);
+      getAllClientData(temp)
+        .then((res) => {
+          console.log(res.data.msg);
+          setClientData(res.data.msg);
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    } else {
+      setToken();
+      setClientData();
+      localStorage.removeItem("token");
     }
-    useEffect(() => {
-        fetchClientData();
-    }, [isLogin]);
+    // console.log("clientData fetched");
+  };
+  useEffect(() => {
+    return () => {
```

Comment:
The useEffect hook is a way to perform side effects in your functional components, such as fetching data, updating the document title, or subscribing to events. The useEffect hook takes a function as an argument, which is called after every render of the component. This function can optionally return another function, which is called before the next render or when the component is unmounted. This returned function is used to clean up any resources or subscriptions that were created by the effect. For example:

// import the useEffect hook
import { useEffect } from "react";

// define a functional component
function Example() {
// use the useEffect hook
useEffect(() => {
// perform some side effect here
console.log("The component has mounted or updated");

// optionally return a cleanup function
return () => {
  // perform some cleanup here
  console.log("The component will unmount or update");
};
});

// return the JSX element
return

Example
;
}